services/datamanager/datasync/sync.go

```diff
@@ -253,6 +276,34 @@ func exponentialRetry(cancelCtx context.Context, fn func(cancelCtx context.Conte
 	}
 }
 
+// isRetryableGRPCError returns true if we should retry syncing and otherwise
+// returns false so that the data gets moved to the corrupted data directory.
+func isRetryableGRPCError(err error) bool {
+	errStatus := status.Convert(err)
+	return errStatus.Code() != codes.InvalidArgument
+}
+
+// moveCorruptedData takes any corrupted data in the captureDir and moves it
+// to a new subdirectory "corrupted" that will not be synced.
+func moveCorruptedData(path, captureDir string) error {
+	// Remove the captureDir part of the path to the corrupted data
+	relativePath, err := filepath.Rel(captureDir, path)
+	if err != nil {
+		return errors.Wrap(err, "error getting relative path of corrupted data")
+	}
+	// Create a new directory captureDir/corrupted/pathToFile
+	newDir := filepath.Join(captureDir, CorruptedDir, filepath.Dir(relativePath))
+	if err := os.MkdirAll(newDir, 0o700); err != nil {
```

Comment:
I think because we likely want to maintain the subdirectory structure of the corrupted data (component name/method), we want to have this here instead of the constructor.